benchmarks/perf-tool/okpt/test/steps/steps.py

```diff
@@ -414,6 +471,134 @@ def _get_measures(self) -> List[str]:
 
         return measures
 
+class QueryWithFilterStep(OpenSearchStep):
```

Comment:
Seems like we are duplicating a lot of code in this method with the query step. Would it be possible to refactor to share common functionality with QueryStep through either helper methods or this class extending QueryStep class?

Same applies to ingest step.

Reply:
agree, let me think more, this should be possible

Reply:
refactored both ingestion and query areas